File: src/uniform-info.ts
export default class UniformInfo<T>{
    name: string
    position: WebGLUniformLocation
    mapper: (webGlContext: WebGLRenderingContext, position: WebGLUniformLocation, data: T) => void
    constructor(
        name: string,
        position: WebGLUniformLocation,
        mapper: (webGlContext: WebGLRenderingContext, position: WebGLUniformLocation, data: T) => void
    ){
        this.name = name
        this.position = position
        this.mapper = mapper
    }
}

File: src/sgl/UniformInfo.kt
package sgl

import org.khronos.webgl.WebGLRenderingContext
import org.khronos.webgl.WebGLUniformLocation

typealias UniformMapper<T> = (renderingContext: WebGLRenderingContext, position: WebGLUniformLocation, data: T) -> Unit

data class UniformInfo<in T>(
        val name: String,
        val position: WebGLUniformLocation,
        val mapper: UniformMapper<T>
)

data class Uniform<in T>(
        val name: String,
        val mapper: UniformMapper<T>
)
